src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,138 @@
+import {
+    Component,
+    Input,
+    ViewEncapsulation,
+    ContentChildren,
+    QueryList,
+    Renderer2,
+    ElementRef,
+    Optional,
+    SkipSelf,
+    ContentChild,
+    Output,
+    AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    },
+    encapsulation: ViewEncapsulation.None
+})
+export class NavigationItemComponent implements AfterViewInit {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link: string;
+
+    @Output()
+    get level(): number {
+        return this._level;
+    }
+
+    @Output()
+    get active() {
+        if (this.link) {
+            return this._router.isActive(this.link, true);
+        }
+    }
+
+    private _level: number = 1;
+    private _active: boolean = false;
+
+    @ContentChildren(NavigationItemComponent, { descendants: true })
+    private _children: QueryList<NavigationItemComponent>;
+
+    get children(): NavigationItemComponent[] {
+        return this._children.filter(item => item !== this);
+    }    
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _renderer: Renderer2,
+        @Optional()
+        @SkipSelf()
+        private _parent: NavigationItemComponent,
+        @Optional() private _router: Router
+    ) {
+        this._level = _parent ? _parent.level + 1 : 1;
+
+        _router.events.filter(event => event instanceof NavigationEnd).subscribe(event => {              
```

Comment:
You need to store the subscription of this and dispose it when the component is destroyed - look into ngOnDestroy lifecycle hook

Reply:
Subscription now stored and disposed of in ngOnDestroy().